src/components/StakingVaults/PositionTable.tsx

```diff
@@ -0,0 +1,177 @@
+      },
+      {
+        Header: translate('defi.netApy'),
+        accessor: 'netApy',
+        textAlign: 'right',
+        Cell: ({ row }: { row: RowProps }) => (
+          <Flex justifyContent={{ base: 'flex-end', md: 'flex-start' }}>
+            <Tag colorScheme='green' size={{ base: 'sm', md: 'md' }}>
+              <Amount.Percent value={row.original.netApy} />
+            </Tag>
+          </Flex>
+        ),
+      },
+      {
+        Header: translate('defi.claimableRewards'),
+        accessor: 'fiatRewardsAmount',
+        display: { base: 'none', md: 'table-cell' },
+        Cell: ({ row }: { row: RowProps }) => {
+          const hasRewards = bnOrZero(row.original.fiatRewardsAmount).gt(0)
+          return hasRewards ? (
+            <Amount.Fiat value={row.original.fiatRewardsAmount} />
+          ) : (
+            <RawText variant='sub-text'>-</RawText>
+          )
+        },
+      },
+      {
+        Header: () => null,
+        id: 'expander',
+        textAlign: 'right',
+        display: { base: 'none', md: 'table-cell' },
+        Cell: ({ row }: { row: RowProps }) => (
+          <Flex justifyContent='flex-end' width='full'>
+            <IconButton
+              variant='ghost'
+              size='md'
+              aria-label='Exapnd Row'
+              isActive={row.isExpanded}
+              icon={row.isExpanded ? <ArrowUpIcon /> : <ArrowDownIcon />}
+            />
+          </Flex>
+        ),
+      },
+    ],
+    [translate],
+  )
+
+  const filterRowsBySearchTerm = useCallback(
+    (rows: AggregatedOpportunitiesByAssetIdReturn[], filterValue: any) => {
+      if (filterValue === '' || filterValue === null || filterValue === undefined) return rows
+      if (typeof filterValue !== 'string') {
+        return []
+      }
+      const search = filterValue.trim().toLowerCase()
+      if (isEthAddress(filterValue)) {
+        return rows.filter(
+          row => fromAssetId(row.assetId).assetReference.toLowerCase() === filterValue,
+        )
+      }
+      const assetIds = rows.map(row => row.assetId)
+      const rowAssets = assets.filter(asset => assetIds.includes(asset.assetId))
+      const matchedAssets = matchSorter(rowAssets, search, { keys: ['name', 'symbol'] }).map(
+        asset => asset.assetId,
+      )
+      const results = rows.filter(row => matchedAssets.includes(row.assetId))
+      return results
+    },
+    [assets],
+  )
+
+  const searching = useMemo(() => searchQuery.length > 0, [searchQuery])
+
+  const rows = useMemo(() => {
+    return searching ? filterRowsBySearchTerm(positions, searchQuery) : positions
+  }, [filterRowsBySearchTerm, positions, searchQuery, searching])
+
+  return (
+    <>
+      {headerComponent &&
+        headerComponent({ setGlobalFilter: setSearchQuery, globalFilter: searchQuery })}
+      <ReactTable
+        onRowClick={row => row.toggleRowExpanded()}
+        data={rows}
+        columns={columns}
+        renderSubComponent={PositionDetails}
```

Comment:
If you want to make `<PositionDetails />` more declarative instead of destructuring `original` in it and tightly coupling it to react-table
```suggestion
        renderSubComponent={({original}) => <PositionDetails {...original}/>}
```